akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -792,8 +792,8 @@ trait FlowOps[+Out, +Mat] {
    */
   def groupedWithin(n: Int, d: FiniteDuration): Repr[immutable.Seq[Out]] = {
     require(n > 0, "n must be greater than 0")
-    require(d > Duration.Zero)
-    via(new GroupedWithin[Out](n, d).withAttributes(name("groupedWithin")))
+    require(d > Duration.Zero) // FIXME shouldn't the requirements of GroupedWithin be verified *inside* GroupedWithin?
```

Comment:
While there are not clear cut answers here, I think it is worth to check most of the things here. My reasoning is that if you check at stage initialization time, you basically allow creating an invalid blueprint that will always result in a faulty stage. Error reports should happen before construction time if the errors are such that would guarantee a faulty blueprint as a result.